### PR TITLE
feat: helm chart support private registry

### PR DESCRIPTION
**Pull request by alifiroozi80**
Hello everyone,
Based on [this feature request](https://github.com/datreeio/admission-webhook-datree/issues/308), I added an `imagePullSecrets` field to the Datree charts.
The Datree chart supports the `registry` field but hasn't had an `imagePullSecrets` field (What if the user had its own registry?)
Thanks to @eyarz and @myishay for encouraging me to open this PR.

### CODE REVIEW

**Comment by myishay** [comment]
Hi @alifiroozi80 ! Thank you for your contribution! 🚀 
One of our team will review it in the upcoming days 👀

**Comment by alifiroozi80** [comment]
Hello @TzlilSwimmer123, 
I applied all suggestions.
Thanks for your feedback.

**Review by TzlilSwimmer123 — COMMENTED** [review]
@alifiroozi80 left 2 last comments.
Did you had a chance to test it with private repo?

**Comment by TzlilSwimmer123** [comment]
@alifiroozi80 in addition, I tried to install it with your changes but something with the helper method doesn't work.
```
Error: INSTALLATION FAILED: parse error at (datree-admission-webhook/templates/_helpers.tpl:27): comment ends before closing delimiter
helm.go:84: [debug] parse error at (datree-admission-webhook/templates/_helpers.tpl:27): comment ends before closing delimiter
INSTALLATION FAILED
main.newInstallCmd.func2
	helm.sh/helm/v3/cmd/helm/install.go:127
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v1.3.0/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.3.0/command.go:974
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v1.3.0/command.go:902
main.main
	helm.sh/helm/v3/cmd/helm/helm.go:83
runtime.main
	runtime/proc.go:255
runtime.goexit
	runtime/asm_arm64.s:1133
make: *** [helm-install-staging] Error 1
```

File: charts/datree-admission-webhook/templates/_helpers.tpl [hunk 1]
```diff
@@ -23,3 +23,10 @@ helm.sh/chart: {{ template "datree.chart" . }}
 {{- define "datree.namespace" -}}
 {{- default .Release.Namespace .Values.namespace  -}}
 {{- end -}}
+
+{{/* The imagePullSecret */}
+{{- define "imagePullSecret" }}
+{{- with .Values.image.imageCredentials }}
+{{- printf "{\"auths\":{\"%s\":{\"username\":\"%s\",\"password\":\"%s\",\"email\":\"%s\",\"auth\":\"%s\"}}}" .registry .username .password .email (printf "%s:%s" .username .password | b64enc) | b64enc }}
+{{- end }}
+{{- end }}
```
Comment (TzlilSwimmer123) [inline charts/datree-admission-webhook/templates/_helpers.tpl:32]: When installing the helm I get 
```
Error: INSTALLATION FAILED: parse error at (datree-admission-webhook/templates/_helpers.tpl:27): comment ends before closing delimiter
helm.go:84: [debug] parse error at (datree-admission-webhook/templates/_helpers.tpl:27): comment ends before closing delimiter
INSTALLATION FAILED
main.newInstallCmd.func2
	helm.sh/helm/v3/cmd/helm/install.go:127
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v1.3.0/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.3.0/command.go:974
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v1.3.0/command.go:902
main.main
	helm.sh/helm/v3/cmd/helm/helm.go:83
runtime.main
	runtime/proc.go:255
runtime.goexit
	runtime/asm_arm64.s:1133
make: *** [helm-install-staging] Error 1
```

In addition `.Values.image.imageCredentials` is incorrect, `imageCredentials` is not inside `image`

Comment (alifiroozi80) [reply charts/datree-admission-webhook/templates/_helpers.tpl]: Sorry for the inconvenience. I fixed that :)
I think now it should be installed fine 👍

**Comment by TzlilSwimmer123** [comment]
I left 2 comments before:)
https://github.com/datreeio/admission-webhook-datree/pull/328#pullrequestreview-1435498037

Did you managed to install the helm chart with private registry?

**Comment by alifiroozi80** [comment]
Hello again, @TzlilSwimmer123.
I just fixed another thing and Installed the Chart successfully.

**Comment by TzlilSwimmer123** [comment]
@alifiroozi80 there is a chance you forgot to push?

**Comment by alifiroozi80** [comment]
No, I pushed the [Fix the helper](https://github.com/datreeio/admission-webhook-datree/pull/328/commits/457028bd5dc734a4320160648dcea7021ac470d1) commit.

With this commit and this sample `values.yaml` file like the one below, it would be installed:

```yaml
datree:
  token: "xxx"
  clusterName: "xxx"

imageCredentials:
  enabled: true
  registry: xxx
  username: xxx
  password: xxx
  email: xxx@xxx.com
```

---

```shell
helm install xxx --dry-run --debug .
```

**Comment by TzlilSwimmer123** [comment]
@alifiroozi80 
there are 2 things to change plz
check this pr https://github.com/datreeio/admission-webhook-datree/pull/328#pullrequestreview-1435498037

**Comment by alifiroozi80** [comment]
Hello @TzlilSwimmer123
Yes, I install it :)

Here are my `values.yaml`:

```yaml
namespace: "test"
datree:
  token: "xxx"
  clusterName: "xxx"

image:
  repository: alifiroozizamani/datree
  tag: "1"

imageCredentials:
  enabled: true
  registry: docker.io
  username: alifiroozizamani
  password: "SuperSecretPassword"
  email: alifiroozizamani@gmail.com

clusterScanner:
  image:
    repository: alifiroozizamani/datree
    tag: "1"
```

---

```shell
helm install -n test test .
```

---

```shell

$ k -n test describe pod datree-webhook-server-7d9f479ff5-4s4tt
[...]
Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   Scheduled  30s                default-scheduler  Successfully assigned test/datree-webhook-server-7d9f479ff5-4s4tt to worker2
  Normal   Pulled     18s                kubelet            Successfully pulled image "alifiroozizamani/datree:1" in 11.185828103s
  Normal   Pulling    16s                kubelet            Pulling image "alifiroozizamani/datree:1"
  Normal   Created    15s                kubelet            Created container server
  Normal   Started    15s                kubelet            Started container server
[...]
```

**Comment by TzlilSwimmer123** [comment]
@alifiroozi80 Maybe there is a problem with the review url so here is a print screen:
Screen Shot 2023-05-21 at 15 06 19

**Comment by alifiroozi80** [comment]
Hello @TzlilSwimmer123,
Thanks, 
I change the `README.md` file.
But I didn't realize the second comment about the `image-credentials.yaml`, (`Forgot to remove new line:)`)

---

Did you mean to add a new line at the end of the `image-credentials.yaml`?

**Review by TzlilSwimmer123 — APPROVED** [review]
nice!

**Comment by TzlilSwimmer123** [comment]
@alifiroozi80 `Did you mean to add a new line at the end of the` - yes

**Comment by alifiroozi80** [comment]
Hello @TzlilSwimmer123, 
Done✅

**Review by TzlilSwimmer123 — APPROVED** [review]
Very cool!

**Comment by alifiroozi80** [comment]
Thank @TzlilSwimmer123 :)

**Pull request merged**